lib/signs_ui/alerts/events.ex

```diff
@@ -0,0 +1,58 @@
+defmodule SignsUi.Alerts.Events do
+  @moduledoc """
+  Parses incoming alert data from the API into multi-route alerts for internal
+  storage.
+  """
+  require Logger
+  alias SignsUi.Alerts.Alert
+
+  @spec parse(String.t() | list() | map()) :: [Alert.multi_route()] | Alert.multi_route()
+  def parse(payload) when is_binary(payload) do
+    payload |> Jason.decode!() |> parse()
+  end
+
+  def parse(payload) when is_list(payload) do
+    Enum.map(payload, &parse/1)
+  end
+
+  def parse(payload) when is_map(payload) do
+    id = payload["id"]
+
+    {created_at, service_effect, routes} = parse_attributes(payload["attributes"])
+
+    %{
```

Comment:
Returning a map rather than an `Alert` (honestly, Dialyzer, what would you even say you do here....? 😞 ).